task/mgmt-gateway/src/main.rs

```diff
@@ -200,13 +221,27 @@ impl UsartHandler {
         // Clear any errors.
         if self.usart.check_and_clear_rx_overrun() {
             ringbuf_entry!(Log::UsartRxOverrun);
+            // TODO-correctness Should we notify MGS of dropped data here? We
+            // could increment `self.from_rx_offset`, but (a) we don't know how
+            // much data we lost, and (b) it would indicate lost data in the
+            // wrong place (i.e., data lost at the current
+            // `self.from_rx_offset`, instead of `self.from_rx_offset +
+            // self.from_rx.len()`, which is where we actually are now).
         }
 
         // Recv as much as we have space for.
         let mut n = 0;
         let mut available_rx_space =
             self.from_rx.capacity() - self.from_rx.len();
 
+        // Ensure we always have space for at least 16 bytes (the stm32h7 usart
+        // FIFO depth), even if that means discarding old data.
+        if available_rx_space < 16 {
```

Comment:
Can we log an error here, even though MGS will also notice it?

Reply:
By "log" do you mean a ringbuf entry or something else?